src/main/scala/gameengine/GameEngine.scala

```diff
@@ -0,0 +1,109 @@
+package gameengine
+
+import characters.{GameBoard, Person, Player, Resources}
+
+import scala.Console.println
+import scala.collection.mutable.ListBuffer
+import scala.io.StdIn.readLine
+import scala.util.Random
+
+class GameEngine {
+  val resources = new Resources
+  private var questionList: List[String] = resources.listOfQuestions
+  private val random = new Random()
+  // Instantiate the game board
+  val gameBoard = new GameBoard(resources.charactersList)
+  // create firstPlayer
+  private val firstPlayer: Player = createPlayerAndAssignGameBoard()
+  private var continuedPlaying: Boolean = true
+
+  /**
+   * Create an object of player by assigning name, gameBoard and secret characters
+   * @return player object
+   */
+  def createPlayerAndAssignGameBoard(): Player = {
+    val player = new Player(name = "Player1", gameBoard = gameBoard.gameBoardForPlayer,
+      selectRandomCharacter(resources.charactersList))
+    player
+  }
+
+
+  /**
+   * Select random character by from Character List from Resources class
+   * @param characterList list of characters from resources class
+   * @return randomCharacter of type Person
+   */
+  def selectRandomCharacter(characterList: List[Person]): Person = {
+    val randomCharacter = characterList(random.nextInt(characterList.length))
+    randomCharacter
+  }
+
+  /**
+   * Select random question from the question list and filters out
+   * questionList which is returned so that there is no repeated questions
+   * @return if
+   *         * list is empty returns there are `no more questions`
+   *         else returns the selected random question
+   */
+  def selectRandomQuestions(): String = {
+    val questionsLength = random.nextInt(questionList.length)
+    val question = questionList(questionsLength)
+    questionList = questionList.filterNot(_ == question)
+    question
+  }
+
+  /** *
+   * Prints the secret Character of player in formatted order
+   * @param secretCharacter the gameBoard assigned for player1
+   */
+  def printSecretCharacterForPlayer(secretCharacter: Person): String = {
+    // Create header for displaying the attribute
+    val header = f"|${"Name"}%-15s | ${"Gender"}%-10s | ${"Hair Colour"}%-15s | ${"Wears Glasses"}%-15s |" +
+      f"|${"Wears Hat"}%-15s ||${"Has Beard"}%-15s ||${"Eye Color"}%-15s |"
+    // create a separator line
+    val separator = "|" + "-" * 17 + "+" + "-" * 12 + "+" + "-" * 17 + "+" + "-" * 17 + "|" + "|" + "-" * 17 + "+" + "-" * 12 + "+" + "-" * 17 + "+" + "-" * 17 + "|"
+    // create the values row
+    val row = f"| ${secretCharacter.name}%-15s | ${secretCharacter.gender}%-10s | ${secretCharacter.hairColor}%-15s | ${secretCharacter.wearsGlasses}%-15s |" +
```

Comment:
Really thought about the layout here for the user, well done!

Reply:
Thanks